web_app/db/crud.py

```diff
@@ -512,3 +512,36 @@ def delete_telegram_user(self, telegram_id: str) -> None:
         user = self.get_user_by_telegram_id(telegram_id)
         if user:
             self.delete_object(user, user.id)
+
+    def set_notification_allowed(self, telegram_id: str = None, wallet_id: str = None, is_allowed: bool = None) -> TelegramUser:
+        """
+        Toggles or sets is_allowed_notification for a TelegramUser, creating a new user if none exists.
+        Either telegram_id or wallet_id must be provided.
+        
+        :param telegram_id: str, optional
+        :param wallet_id: str, optional
+        :param is_allowed: bool, optional - if not provided, toggles the current value
+        :return: TelegramUser
+        """
+        if not telegram_id and not wallet_id:
+            raise ValueError("Either telegram_id or wallet_id must be provided")
+
+        with self.Session() as session:
+            user = None
+            if telegram_id:
+                user = self.get_user_by_telegram_id(telegram_id)
+            if not user and wallet_id:
+                user = session.query(TelegramUser).filter_by(wallet_id=wallet_id).first()
+
+            if user:
+                user.is_allowed_notification = not user.is_allowed_notification if is_allowed is None else is_allowed
+                session.commit()
+                session.refresh(user)
+                return user
+            else:
+                user_data = {
+                    "telegram_id": telegram_id,
+                    "wallet_id": wallet_id,
+                    "is_allowed_notification": True if is_allowed is None else is_allowed
```

Comment:
This code is broken, you didn't define `is_allowed` variable, so it's potential buggy place.
if user is not exist, `is_allowed` should be True